### PR TITLE
bug fix in unversion request

### PR DESCRIPTION
**Pull request by magicwang-cn**
ref #20136 

@lavalamp @liggitt @kevin-wangzefeng @alfred-huangjian @HardySimpson

### CODE REVIEW

File: pkg/client/unversioned/request.go [hunk 1]
```diff
@@ -642,6 +642,7 @@ func (r *Request) Watch() (watch.Interface, error) {
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
+		defer resp.Body.Close()
```
Comment (lavalamp) [inline pkg/client/unversioned/request.go]: Protect with a check for nil, don't remove it entirely!

Comment (lavalamp) [inline pkg/client/unversioned/request.go]: Nevermind. I am totally confused :)

Comment (caesarxuchao) [inline pkg/client/unversioned/request.go]: Why not doing it inside the transformResponse?

Comment (magicwang-cn) [inline pkg/client/unversioned/request.go]: yes. i think of it. i wonder whether it(`the resp`) will be  used out side of the transformResponse.

Comment (caesarxuchao) [inline pkg/client/unversioned/request.go]: It seems `resp.Body` is not used out side of transformResponse. Could you move it to where it's used inside transformResponse? Thanks.

Comment (liggitt) [inline pkg/client/unversioned/request.go]: I'd rather keep it here to make it easy to confirm that the body is closed in the same place we create it

Comment (caesarxuchao) [inline pkg/client/unversioned/request.go]: My bad, my understanding of `Close()` was wrong, we should close the body in this function. Although shall we do it out of the if condition, right after the error check?

Comment (liggitt) [inline pkg/client/unversioned/request.go]: no, otherwise the returned streamwatcher would be useless, since the underlying body would have been closed.

Comment (caesarxuchao) [inline pkg/client/unversioned/request.go]: I see. Thanks for the explanation. LGTM.

**Pull request merged**